mock/service/node.go

```diff
@@ -254,12 +260,22 @@ func (s *service) NodeUnpublishVolume(
 		// nodeMntPathKey is the key in the volume's attributes that is set to a
 		// mock mount path if the volume has been published by the node
 		nodeMntPathKey := path.Join(s.nodeID, req.TargetPath)
+		nodeMntPathCreated := nodeMntPathKey + "Created"
 
 		// Check to see if the volume has already been unpublished.
 		if v.VolumeContext[nodeMntPathKey] == "" {
 			return &csi.NodeUnpublishVolumeResponse{}, nil
 		}
 
+		// Check to see if we created the volume, and if so delete it
+		if v.VolumeContext[nodeMntPathCreated] == "true" {
```

Comment:
do we need this new key or should we always delete?

Reply:
I was initially hesitant on always deleting, but considering that this driver is used for testing I don't think there's an issue. I've adjusted this section to always delete the targetPath.